backend/fms_core/template_importer/importers/sample_pooling.py

```diff
@@ -0,0 +1,79 @@
+from fms_core.models import Protocol
+from ._generic import GenericImporter
+from fms_core.template_importer.row_handlers.sample_pooling import SamplesToPoolRowHandler, PoolsRowHandler
+from fms_core.templates import SAMPLE_POOLING_TEMPLATE
+from collections import defaultdict
+from .._utils import float_to_decimal_and_none, input_to_date_and_none
+from fms_core.utils import str_cast_and_normalize, str_cast_and_normalize_lower, check_truth_like
+
+class SamplePoolingImporter(GenericImporter):
+    SHEETS_INFO = SAMPLE_POOLING_TEMPLATE['sheets info']
+
+    def __init__(self):
+        super().__init__()
+        self.initialize_data_for_template()
+
+    def initialize_data_for_template(self):
+        self.preloaded_data = {'protocol': Protocol.objects.get(name='Sample Pooling')}
+
+    def import_template_inner(self):
+        pools_dict = defaultdict(list)
+        """
+            SamplesToPool SHEET
+        """
+        samplestopool_sheet = self.sheets['SamplesToPool']
+        for i, row_data in enumerate(samplestopool_sheet.rows):
+            samplestopool_kwargs = {
+                'source_sample':
+                    {'barcode': str_cast_and_normalize(row_data['Source Container Barcode']),
+                     'coordinates': str_cast_and_normalize(row_data['Source Container Coord']),
+                     'depleted': check_truth_like(row_data['Source Depleted']) if row_data['Source Depleted'] else None,
+                     },
+                'volume_used': float_to_decimal_and_none(row_data['Volume Used (uL)']),
+                'comment': str_cast_and_normalize(row_data['Comment']),
+            }
+
+            (result, row_object) = self.handle_row(
+                row_handler_class=SamplesToPoolRowHandler,
+                sheet=samplestopool_sheet,
+                row_i=i,
+                **samplestopool_kwargs,
+            )
+            pools_dict[str_cast_and_normalize(row_data['Pool Name'])].append(row_object)
+
+        if not result['validation_error'].messages: # prevent the repetition of error messages for volume_used at the level of pools.
```

Comment:
This will crash because result is initialized inside the loop, so here it is referenced before assignment